src/foundry/common/primitives/math.mjs.d.ts

```diff
@@ -61,4 +61,15 @@ interface Math {
    * @returns An angle in radians
    */
   toRadians(angle: number): number;
+
+  /**
+   * Get an oscillation between lVal and hVal according to t
+   * @param  minVal - The minimal value of the oscillation.
+   * @param  maxVal - The maximum value of the oscillation.
+   * @param  t      - The time value.
+   * @param  p      - The period (can't be equal to 0).
+   * @param  func   - The optional math function to use for oscillation.
+   * @returns The oscillation according to t.
+   */
+  oscillation(minVal: number, maxVal: number, t: number, p: number = 1, func: Function = Math.cos): number;
```

Comment:
These default values won't work. You're not being warned because of `skipLibCheck` but you can't set defaults in signatures unless they're actual functions, unfortunately.

If you unset `skipLibCheck` you should see this error:
```
A parameter initializer is only allowed in a function or constructor implementation.(2371)
```

You can add a line to the documentation like ``(default: `1`)`` and ``(default = `Math.cos`)`` and remove them. All in all it'd look like so:
```suggestion
  /**
   * Get an oscillation between lVal and hVal according to t
   * @param  minVal - The minimal value of the oscillation.
   * @param  maxVal - The maximum value of the oscillation.
   * @param  t      - The time value.
   * @param  p      - The period (can't be equal to 0).
   *                  (default: `1`)
   * @param  func   - The optional math function to use for oscillation.
   *                  (default: `Math.cos`)
   * @returns The oscillation according to t.
   */
  oscillation(minVal: number, maxVal: number, t: number, p?: number, func?: (radians: number) => number): number;
```